packages/amesos/test/Test_Basic/Amesos_TestSolver.cpp

```diff
@@ -225,7 +231,11 @@ int Amesos_TestSolver( Epetra_Comm &Comm, char *matrix_file,
     Comm.Barrier();
     
     A.Export(*serialA, exporter, Add);
-    assert(A.FillComplete()==0);    
+    //assert(A.FillComplete()==0);
+    if(A.FillComplete()!=0) {
+      exit(0);
+    }
+    delete serialA;
```

Comment:
Does this need to be guarded by `if (distribute_matrix)` to balance the case below around L522? Why is this deleted here but conditionally deleted at L522, breaking the convention of the other items?

Reply:
@ndellingwood. This is guarded by `if (distribued_matrix)' on L211? I am having challenge running this test with a large matrix, and was hoping freeing up memory will help.